src/joystick/linux/SDL_sysjoystick.c

```diff
@@ -317,43 +318,17 @@ static int IsJoystick(const char *path, int fd, char **name_return, SDL_Joystick
 static int IsSensor(const char *path, int fd)
 {
     struct input_id inpid;
-    char *name;
-    char product_string[128];
 
     if (ioctl(fd, EVIOCGID, &inpid) < 0) {
         return 0;
     }
 
-    if (ioctl(fd, EVIOCGNAME(sizeof(product_string)), product_string) < 0) {
-        return 0;
-    }
-
-    name = SDL_CreateJoystickName(inpid.vendor, inpid.product, NULL, product_string);
-    if (name == NULL) {
-        return 0;
-    }
-
-    if (SDL_endswith(name, " Motion Sensors")) {
-        /* PS3 and PS4 motion controls */
-        SDL_free(name);
-        return 1;
-    }
-    if (SDL_strncmp(name, "Nintendo ", 9) == 0 && SDL_strstr(name, " IMU") != NULL) {
-        /* Nintendo Switch Joy-Con and Pro Controller IMU */
-        SDL_free(name);
-        return 1;
-    }
-    if (SDL_endswith(name, " Accelerometer") ||
-        SDL_endswith(name, " IR") ||
-        SDL_endswith(name, " Motion Plus") ||
-        SDL_endswith(name, " Nunchuk")) {
+    if (inpid.vendor == USB_VENDOR_NINTENDO && inpid.product == USB_PRODUCT_NINTENDO_WII_REMOTE) {
```

Comment:
Does this properly detect the Wii IMU?
In theory we should be able to use the Accelerometer + Motion Plus, right?

Reply:
> Does this properly detect the Wii IMU?

You mean *Switch* IMU? I do not own a Switch controller and there is no data in `testevdec.c` but the Linux driver sets the accelerometer prop so `SDL_EVDEV_GuessDeviceClass()` should guess correctly.

> In theory we should be able to use the Accelerometer + Motion Plus, right?

Looking at the Linux driver it seems that they do not set `uniq` (https://elixir.bootlin.com/linux/latest/source/drivers/hid/hid-wiimote-modules.c#L495) so we cannot associate them with a joystick. But someone with a Wii controller should check that.

Reply:
Okay, fair enough.